aiida/engine/processes/calcjobs/manager.py

```diff
@@ -7,61 +7,81 @@
 # For further information on the license, see the LICENSE.txt file        #
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
-"""
-Module containing utilities and classes relating to job calculations running
-on systems that require transport.
-"""
+"""Module containing utilities and classes relating to job calculations running on systems that require transport."""
 from __future__ import division
 from __future__ import print_function
 from __future__ import absolute_import
 
 import contextlib
-from functools import partial
 import time
 
 from six import iteritems, itervalues
 from tornado import concurrent, gen
 
 from aiida import schedulers
-from aiida.common import exceptions
-from ...utils import RefObjectStore
+from aiida.common import exceptions, lang
+from aiida.common.log import AIIDA_LOGGER
 
 __all__ = ('JobsList', 'JobManager')
 
 
 class JobsList(object):  # pylint: disable=useless-object-inheritance
-    """
-    A list of submitted jobs on a machine connected to by transport based on the
-    authorisation information.
+    """A list of submitted jobs on a machine connected to by transport based on the authorisation information.
+
+    This container of active calculation jobs is used to update their status periodically in batches. This ensure that
+    even when a lot of jobs are running the scheduler update command is not triggered for each job individually.
+
+    In addition, the `Computer` for which the `AuthInfo` is configured, can define a minimum polling interval. This
+    class will guarantee that the time between update calls to the scheduler is larger or equal to that minimum
+    interval.
+
+    Note that since each instance operates on a specific `AuthInfo`, the guarantees of batching scheduler update calls
```

Comment:
Note that there will be one `JobsList` instance *per daemon worker* and *per `AuthInfo`*.
I.e. when, hypothetically, configuring the same computer for multiple users, and running jobs for different users simultaneously, the minimum polling interval would be respected only *per user*.

Reply:
I explicitly did not reference the "daemon runner" here, because this applies to any runner. Also when you "run" a job, it creates a runner, which has its own job manager and own job lists for each authentication info. Since all these extra conditions have nothing to do with the `JobsList` I did not include them here. This behavior should rather be documented in a more general place. This is already done in the documentation of the `get_minimum_job_poll_interval`.

Reply:
In this case, it still makes sense to mention that it applies to any runner individually.
This is not obvious.

Reply:
But I don't think that is a limitation of this class and therefore should not be in this docstring. This class is simply used by the `JobManager` which then makes sure there is one per `AuthInfo`. However, then still we do not have a guarantee that this is unique per authinfo. A runner could just this instantiate two `JobManager` instances and then the condition that is written in the dosctring of the `JobsList` would be broken. I think it is just the wrong place for this information as it is not enforced here at all.

Reply:
I get your point - I'm just saying that, from a user perspective, the use case of running jobs using multiple authinfos on the same computer is quite esoteric compared to the highly relevant case of having multiple runners. 
So, as a user reading this docstring, that is what I would like it to tell me.

Perhaps one way to resolve this is to mention that the JobList is per JobManager and put a link to the JobManager's  documentation (which then should somehow mention its relation to the runners and what this means for any per-computer limits).

Reply:
I have put a reference to the `JobManager` in the `JobsList` docstring and explained the bigger picture with the runners in the `JobManager` docstring